app/src/main/java/com/example/tasky/common/domain/TextValidatorExtension.kt

```diff
@@ -0,0 +1,17 @@
+package com.example.tasky.common.domain
+
+import android.util.Patterns
+
+
+fun String.isValidPassword(): Boolean {
+    val passwordPattern = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d).{9,}$"
+    return this.matches(passwordPattern.toRegex())
+}
+
+fun String.isValidEmail(): Boolean {
+    return Patterns.EMAIL_ADDRESS.matcher(this).matches()
+}
```

Comment:
Here, you're using an Android-specific dependency (as you can see in the imports) which doesn't really belong into domain. Instead, I'd recommend to create an abstraction for this to keep this class locally unit testable.

Every time, you have implementation details in domain and want to get rid of that, the procedure is the same:

1. You create an abstraction (interface) in domain
2. You write an implementation of that in data (in this case using Patterns.EMAIL_ADDRESS...)
3. You pass the abstraction to the constructor of your class needing it in domain
4. Now you can decide which implementation you pass when you create that class/use case
For example:

*In domain*
```
interface EmailPatternValidator {
    fun isValidEmailPattern(email: String): Boolean
}
```

*In data*
```
class EmailPatternValidatorImpl: EmailPatternValidator {

    override fun isValidEmailPattern(email: String): Boolean {
        return Patterns.EMAIL_ADDRESS.matcher(email).matches()
    }
}
```

Then inject an `EmailPatternValidator` where you want to do the validation.